src/rest/connector/libs/vmware/implementation.py

```diff
@@ -0,0 +1,390 @@
+    @isconnected
+    def post(self, api_url, payload, timeout=30):
+        """POST REST Command to configure new information on the device
+
+        Args:
+            dn (string): Unique distinguished name that describes the object
+                         and its place in the tree.
+
+            payload (dict): Dictionary containing the information to send via
+                            the post
+
+            headers (dict): Headers to send with the rest call
+
+            timeout (int): Maximum time
+
+        Returns:
+            response.json() or response.text
+
+        Raises:
+            RequestException if response is not ok
+        """
+        return self._request('POST', api_url, data=json.dumps(payload), timeout=timeout)
+
+    @BaseConnection.locked
+    @isconnected
+    def delete(self, api_url, timeout=30):
+        """DELETE REST Command to delete information from the device
+
+        Args
+            dn (string): Unique distinguished name that describes the object
+                         and its place in the tree.
+
+            headers (dict): Headers to send with the rest call
+
+            timeout (int): Maximum time
+
+        Returns:
+            response.json() or response.text
+
+        Raises:
+            RequestException if response is not ok
+        """
+        return self._request('DELETE', api_url, timeout=timeout)
+
+    @BaseConnection.locked
+    @isconnected
+    def patch(self, api_url, payload, timeout=30):
+        """PATCH REST Command to partially update existing information on the device
+
+        Args
+            dn (string): Unique distinguished name that describes the object
+                         and its place in the tree.
+
+            payload (dict): Dictionary containing the information to send via
+                            the post
+
+            headers (dict): Headers to send with the rest call
+
+            timeout (int): Maximum time
+
+        Returns:
+            response.json() or response.text
+
+        Raises:
+            RequestException if response is not ok
+        """
+        return self._request('PATCH', api_url, data=json.dumps(payload),
+                             timeout=timeout)
+
+    @BaseConnection.locked
+    @isconnected
+    def put(self, api_url, payload, timeout=30):
+        """PUT REST Command to update existing information on the device
+
+        Args
+            dn (string): Unique distinguished name that describes the object
+                         and its place in the tree.
+
+            payload (dict): Dictionary containing the information to send via
+                            the post
+
+            headers (dict): Headers to send with the rest call
+
+            timeout (int): Maximum time
+
+        Returns:
+            response.json() or response.text
+
+        Raises:
+            RequestException if response is not ok
+        """
+        return self._request('DELETE', api_url, data=json.dumps(payload),
```

Comment:
I think you meant PUT here
```suggestion
        return self._request('PUT', api_url, data=json.dumps(payload),
```